src/catalog/dependency_manager.cpp

```diff
@@ -408,9 +409,58 @@ void DependencyManager::AlterObject(CatalogTransaction transaction, CatalogEntry
 		// It makes no sense to have a schema depend on anything
 		D_ASSERT(dep.EntryInfo().type != CatalogType::SCHEMA_ENTRY);
 
-		throw DependencyException("Cannot alter entry \"%s\" because there are entries that "
-		                          "depend on it.",
-		                          old_obj.name);
+		bool disallow_alter = true;
+		switch (alter_info.type) {
+		case AlterType::ALTER_TABLE: {
+			auto &alter_table = alter_info.Cast<AlterTableInfo>();
+			switch (alter_table.alter_table_type) {
+			case AlterTableType::FOREIGN_KEY_CONSTRAINT: {
+				// These alters are made as part of a CREATE or DROP table statement when a foreign key column is
+				// present either adding or removing a reference to the referenced primary key table
+				disallow_alter = false;
+				break;
+			}
+			case AlterTableType::RENAME_TABLE: {
+				// Renaming of tables shouldnt be blocked by anything
+				disallow_alter = false;
```

Comment:
Let's keep the changeset minimal and add this separately - can we remove the exception for `RENAME_TABLE` for now